File: 2404. Most Frequent Even Element/Solution.py
from collections import Counter
from typing import List


class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        res = -1
        cnt = Counter()
        for x in nums:
            if x % 2:
                continue
            cnt[x] += 1
            if cnt[x] > cnt[res] or cnt[x] == cnt[res] and x < res:
                res = x
        return res

File: 2404. Most Frequent Even Element/Solution.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int mostFrequentEven(vector<int>& nums)
    {
        unordered_map<int, int> cnt;
        int res = -1;
        for (const int& x : nums) {
            if (x & 1)
                continue;
            int c = ++cnt[x];
            if (c > cnt[res] || (c == cnt[res] && x < res))
                res = x;
        }
        return res;
    }
};
